### DISCUSSION TITLE
Enable setting of CRON interval

### DISCUSSION THREAD

**Discussion by EricdeWaal** [akeeba/panopticon#636] [General]
Hi,

Great work!
Many hosting companies do not allow a cron job to run every minute. Mine, for example, only allows a cron task to be executed once an hour. 
This leads to annoying warnings every time when I load a page in Akeeba Panopticon.
can you make it possible to set the check interval?

Best,
Eric

**Comment by nikosdion** [comment]
The requirement for CRON jobs every 1 minute is not just a recommendation, it's a fundamental requirement. If you do not have CRON jobs running every minute you _will_ have problems up to and including broken sites.

To begin with, Joomla! update **will** break if it needs to continue running on another CRON job. Example situations: a. it starts at the end of the CRON job's maximum execution time b. it takes more than the CRON job's maximum execution time. However, Joomla's extract.php file is only unlocked for a _short_ period of time, about 3 minutes. If you have CRON jobs executing every 5 minutes you will end up with a half-updated and 100% broken site.

Even if you don't have a drastic problem like that, you _will_ end up with half-updated sites for the duration of the interval between CRON jobs minus their maximum execution time. With CRON jobs every minute this duration is usually non-existent as the CRON job can run for longer than one minute, and at worst it's 45 seconds.

So, no, I am not going to change the check time (which is currently 3 minutes, mind you!) because anything higher than that is **GUARANTEED** to cause functional problems to your sites, including breaking your site during an update.

My recommendation is that you use a different hosting company, or just set up a cheap Nanode on Akamai (formerly Linode) for $5 a month. This has far more computing power, memory, and bandwidth that you'll reasonably need to tackle a couple hundred sites.

**Reply by EricdeWaal** [reply]
So, if I understand you well, it is not sufficient to trigger a Joomla installation once to start the update process?
Even if the PHP execution time hase been set quite long, it needs to be retriggered by the cron?
And equally for the core and extensions?
It's not a matter of money, I just want to understand the process!

**Reply by nikosdion** [reply]
It depends on the site but, usually, no, one CRON job is not enough for a _core_ update. Let's say you have CRON jobs with a maximum execution time of 60 seconds for the sake of simplicity.

The core update is four parts: download, extract, post-update, cleanup.

The first part runs, downloading the update ZIP file. In most cases, it's big enough and Joomla's Amazon S3 bucket slow enough that this takes about 50 seconds. This is perilously close to the time limit of 60 seconds, therefore the next part needs to be executed by a new CRON job (we don't want a timeout to kill us before we're done!).

The next part is extraction, and it can run in multiple steps, each one taking about 5-10 seconds. With some servers you can extract the whole update package in one 5 second step. On some servers you may have to run 20 or so steps, spanning two minutes i.e. another two CRON jobs.

The last two parts take a few seconds each, so they definitely fit inside a single CRON job.

Now, if you have CRON jobs running every minute the aforementioned process will take exactly as long as it would've taken Joomla itself to run it if you were logged into the admin of the site with your browser. On a slow server it would be about 3 minutes. If your CRON jobs, however, run every five minutes this now takes 15 minutes and change. If your CRON jobs run every ten minutes this takes over half an hour.

Remember, until this is fully done, your site is half-working.

Extension updates are different. Joomla! only exposes a single call to update an extension. It either happens in whole (download, extract, install, cleanup) or it doesn't. The time between CRON jobs won't affect how long it takes a single extension to be updated, but how long it will take _all_ pending extensions to be updated. 

If you have ten extensions to update, each one taking ten seconds to update you would need 2 minutes if your CRON jobs run every minute, but 6 minutes if they run every 5 minutes.

Things get worse because _executing_ updates is not the only thing you ask the CRON jobs to do. You ask them to monitor the sites' uptime, check for updates, run backups etc. If your CRON jobs run once every 5 minutes you can only run at 20% capacity. If you have a dozen site you will see that updates take several days to be discovered and applied, just because there is not enough processing opportunity.

Also read https://github.com/akeeba/panopticon/wiki/CRON-Jobs#busy-installations-will-need-more-than-one-cron-job
